drivers/scheduler/k8s/k8s.go

```diff
@@ -5270,6 +5422,21 @@ func MakePVCFromSnapshot(ns string, name string, sourceSnapshotName string, stor
 	return PVCSpec
 }
 
+// MakePVCByCloning takes namespace, name, source snapshot name and storageclass as parameter and returns a PersistentVolumeClaim spec for PVC creation
+func MakePVCByCloning(size resource.Quantity, ns string, name string, sourcePVCName string, storageClass string) *v1.PersistentVolumeClaim {
+	PVCSpec := MakePVC(size, ns, name, storageClass)
+	PVCSpec.ObjectMeta.Name = name
+	PVCSpec.Namespace = ns
+	PVCSpec.ObjectMeta.Namespace = ns
+	if sourcePVCName != "" {
```

Comment:
We should probably just throw an error if source PVC name is empty... might help prevent some strangeness where it just ends up making a new volume instead